src/main/scala/firrtl/stage/FirrtlCompilerModes.scala

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package firrtl.stage
+
+import firrtl.transforms._
+import firrtl.passes.memlib._
+import firrtl.options.{ShellOption, HasShellOptions}
+
+object OptimizeForFPGA extends HasShellOptions {
+  private val fpgaAnnos = Seq(
+    InferReadWriteAnnotation,
+    RunFirrtlTransformAnnotation(new InferReadWrite),
+    DefaultReadFirstAnnotation,
+    RunFirrtlTransformAnnotation(new SetDefaultReadUnderWrite),
+    RunFirrtlTransformAnnotation(new SimplifyMems),
+    PassthroughSimpleSyncReadMemsAnnotation
```

Comment:
Would this be a good place to explain why these passes are important for FPGA emission?
I feel like this would be the first place that I would look if I wanted to understand what the flag does.
Most people doing that will probably have even less of an understanding of what these passes do than me.

Reply:
Yes, I totally agree. I still need to add a great deal of ScalaDoc and comments, along with expanding the help text.

Reply:
Added